File: process_csv.js
const fs = require('fs');
const csv = require('csv-parser');
const { createObjectCsvWriter } = require('csv-writer');
const { translate } = require('@vitalets/google-translate-api');

function parseImages(imagesStr) {
  if (!imagesStr) return '{}';
  const urls = imagesStr.split(',').map(u => u.trim());
  return '{' + urls.map(u => `"${u}"`).join(',') + '}';
}

function parseFlavors(row) {
  let flavors = [];
  for (let i = 1; i <= 5; i++) {
    const attrName = row[`Nom de l’attribut ${i}`];
    const attrVal = row[`Valeur(s) de l’attribut ${i} `] || row[`Valeur(s) de l’attribut ${i}`];
    if (attrName && attrName.toUpperCase().includes('PARFUM') && attrVal) {
      flavors = flavors.concat(attrVal.split(',').map(s => s.trim()));
    }
  }
  if (flavors.length === 0) return '{}';
  return '{' + flavors.map(f => `"${f.replace(/"/g, '""')}"`).join(',') + '}';
}

async function doTranslate(text) {
  if (!text) return '';
  try {
    const res = await translate(text, { to: 'en' });
    return res.text;
  } catch (err) {
    return text;
  }
}

function generateSlug(name) {
  return name.toLowerCase().replace(/[^a-z0-9]+/g, '-').replace(/(^-|-$)+/g, '');
}

async function main() {
  const results = [];
  console.log('Reading CSV...');
  
  await new Promise((resolve) => {
    fs.createReadStream('wc_product_nutrifitness.csv')
      .pipe(csv())
      .on('data', (data) => results.push(data))
      .on('end', () => resolve());
  });

  console.log(`Found ${results.length} rows.`);

  const mappedData = [];

  for (let i = 0; i < results.length; i++) {
    const row = results[i];
    console.log(`Processing ${i + 1}/${results.length}: ${row.Nom}`);

    let name = row.Nom || '';
    let descShort = row['Description courte'] || '';
    let descLong = row.Description || '';
    let category = row['Catégories'] || '';
    
    name = await doTranslate(name);
    descShort = await doTranslate(descShort);
    descLong = await doTranslate(descLong);
    category = await doTranslate(category);

    const price = row['Tarif promo'] || row['Tarif régulier'] || '0';
    const priceOrig = row['Tarif promo'] ? row['Tarif régulier'] : '';
    
    const stockQuantity = parseInt(row.Stock) || 0;
    const inStock = stockQuantity > 0 || row['En stock ?'] === '1' ? true : false;
    const isFeatured = row['Mis en avant ?'] === '1' ? true : false;

    mappedData.push({
      sku: row.UGS || `SKU-${row.ID}`,
      name: name,
      brand: row.Marques || '',
      slug: generateSlug(name || `product-${row.ID}`),
      description_short: descShort,
      description_long: descLong,
      category: category.split(',')[0]?.trim() || '',
      subcategory: '',
      price: parseFloat(price) || 0,
      price_original: parseFloat(priceOrig) || null,
      currency: 'CHF',
      weight_g: (parseFloat(row['Poids (kg)']) || 0) * 1000,
      flavors: parseFlavors(row),
      images: parseImages(row.Images),
      in_stock: inStock,
      stock_quantity: stockQuantity,
      is_featured: isFeatured,
      is_new: false,
      badge_text: '',
      badge_color: '#c8102e',
      rating: 0,
      review_count: 0,
      tags: '{}'
    });
  }

  console.log('Writing to output CSV...');

  const csvWriter = createObjectCsvWriter({
    path: 'supabase_products.csv',
    header: [
      { id: 'sku', title: 'sku' },
      { id: 'name', title: 'name' },
      { id: 'brand', title: 'brand' },
      { id: 'slug', title: 'slug' },
      { id: 'description_short', title: 'description_short' },
      { id: 'description_long', title: 'description_long' },
      { id: 'category', title: 'category' },
      { id: 'subcategory', title: 'subcategory' },
      { id: 'price', title: 'price' },
      { id: 'price_original', title: 'price_original' },
      { id: 'currency', title: 'currency' },
      { id: 'weight_g', title: 'weight_g' },
      { id: 'flavors', title: 'flavors' },
      { id: 'images', title: 'images' },
      { id: 'in_stock', title: 'in_stock' },
      { id: 'stock_quantity', title: 'stock_quantity' },
      { id: 'is_featured', title: 'is_featured' },
      { id: 'is_new', title: 'is_new' },
      { id: 'badge_text', title: 'badge_text' },
      { id: 'badge_color', title: 'badge_color' },
      { id: 'rating', title: 'rating' },
      { id: 'review_count', title: 'review_count' },
      { id: 'tags', title: 'tags' }
    ]
  });

  await csvWriter.writeRecords(mappedData);
  console.log('Done! Generated supabase_products.csv');
}

main().catch(console.error);

File: process_csv.py
import csv
import re
import sys

def parse_images(images_str):
    if not images_str:
        return '{}'
    urls = [u.strip() for u in images_str.split(',') if u.strip()]
    if not urls:
        return '{}'
    return '{' + ','.join([f'"{u}"' for u in urls]) + '}'

def parse_flavors(row):
    flavors = []
    for i in range(1, 6):
        attr_name = row.get(f'Nom de l’attribut {i}', '')
        attr_val = row.get(f'Valeur(s) de l’attribut {i} ', '') or row.get(f'Valeur(s) de l’attribut {i}', '')
        
        if attr_name and 'PARFUM' in attr_name.upper() and attr_val:
            flavors.extend([f.strip() for f in attr_val.split(',') if f.strip()])
            
    if not flavors:
        return '{}'
    return '{' + ','.join([f'"{f}"' for f in flavors]) + '}'

def generate_slug(name):
    # Basic slug generation
    slug = name.lower()
    slug = re.sub(r'[^a-z0-9]+', '-', slug)
    slug = re.sub(r'(^-|-$)', '', slug)
    return slug

def process_csv(input_file, output_file):
    mapped_data = []
    
    with open(input_file, mode='r', encoding='utf-8-sig') as infile:
        reader = csv.DictReader(infile)
        rows = list(reader)
        print(f"Found {len(rows)} rows.")
        
        for idx, row in enumerate(rows):
            name = row.get('Nom', '')
            # If name is empty and it's a variation, try to use parent or just skip
            if not name and row.get('Type') == 'variation':
                name = f"Variation {row.get('ID')}"
                
            desc_short = row.get('Description courte', '')
            desc_long = row.get('Description', '')
            category = row.get('Catégories', '').split(',')[0].strip() if row.get('Catégories') else ''
            
            # Translate basic categories from French to English
            cat_translations = {
                'APRES SPORT': 'Post Workout',
                'AVANT SPORT': 'Pre Workout',
                'PENDANT SPORT': 'Intra Workout',
                'PERTE DE POIDS': 'Weight Loss',
                'PRISE DE MASSE': 'Mass Gainer',
                'PROTÉINES': 'Proteins',
                'VITAMINES - MINÉRAUX': 'Vitamins & Minerals',
                'ACCESSOIRES': 'Accessories',
                'SHAKERS': 'Shakers',
                'NOS BEST-SELLERS': 'Best Sellers'
            }
            category_en = cat_translations.get(category.upper(), category)

            price = row.get('Tarif promo') or row.get('Tarif régulier') or '0'
            price_orig = row.get('Tarif régulier') if row.get('Tarif promo') else ''
            
            stock_str = row.get('Stock', '0')
            try:
                stock_quantity = int(float(stock_str)) if stock_str else 0
            except ValueError:
                stock_quantity = 0
                
            in_stock = stock_quantity > 0 or row.get('En stock ?') == '1'
            is_featured = row.get('Mis en avant ?') == '1'
            
            weight_str = row.get('Poids (kg)', '0')
            try:
                weight_g = int(float(weight_str) * 1000) if weight_str else 0
            except ValueError:
                weight_g = 0

            mapped_data.append({
                'sku': row.get('UGS') or f"SKU-{row.get('ID')}",
                'name': name,
                'brand': row.get('Marques', ''),
                'slug': generate_slug(name or f"product-{row.get('ID')}"),
                'description_short': desc_short,
                'description_long': desc_long,
                'category': category_en,
                'subcategory': '',
                'price': price,
                'price_original': price_orig,
                'currency': 'CHF',
                'weight_g': weight_g,
                'flavors': parse_flavors(row),
                'images': parse_images(row.get('Images')),
                'in_stock': str(in_stock).lower(),
                'stock_quantity': stock_quantity,
                'is_featured': str(is_featured).lower(),
                'is_new': 'false',
                'badge_text': '',
                'badge_color': '#c8102e',
                'rating': 0,
                'review_count': 0,
                'tags': '{}'
            })
            
    if not mapped_data:
        print("No data parsed.")
        return

    # Write to new CSV
    fieldnames = [
        'sku', 'name', 'brand', 'slug', 'description_short', 'description_long',
        'category', 'subcategory', 'price', 'price_original', 'currency', 'weight_g',
        'flavors', 'images', 'in_stock', 'stock_quantity', 'is_featured', 'is_new',
        'badge_text', 'badge_color', 'rating', 'review_count', 'tags'
    ]
    
    with open(output_file, mode='w', encoding='utf-8', newline='') as outfile:
        writer = csv.DictWriter(outfile, fieldnames=fieldnames)
        writer.writeheader()
        for data in mapped_data:
            writer.writerow(data)
            
    print(f"Successfully generated {output_file} with {len(mapped_data)} products!")

if __name__ == '__main__':
    process_csv('wc_product_nutrifitness.csv', 'supabase_products.csv')
